src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php

```diff
@@ -75,37 +85,28 @@ public function register_routes() {
 	}
 
 	/**
-	 * Checks if the user has permissions to make the request.
-	 *
-	 * @since 5.9.0
+	 * Checks if a given request has access to read a single global styles config.
 	 *
-	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return bool|WP_Error True if the request has read access for the item, WP_Error object otherwise.
 	 */
-	protected function permissions_check() {
-		// Verify if the current user has edit_theme_options capability.
-		// This capability is required to edit/view/delete templates.
-		if ( ! current_user_can( 'edit_theme_options' ) ) {
+	public function get_item_permissions_check( $request ) {
+		$post = $this->get_post( $request['id'] );
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		if ( 'edit' === $request['context'] && $post && ! $this->check_update_permission( $post ) ) {
 			return new WP_Error(
-				'rest_cannot_manage_global_styles',
-				__( 'Sorry, you are not allowed to access the global styles on this site.' ),
-				array(
-					'status' => rest_authorization_required_code(),
-				)
+				'rest_forbidden_context',
+				__( 'Sorry, you are not allowed to edit this global style.' ),
+				array( 'status' => rest_authorization_required_code() )
 			);
 		}
 
-		return true;
+		return current_user_can( 'read_post', $post->ID );
```

Comment:
Can we return a `WP_Error` object here, that way they get the more descriptive error.

Reply:
@TimothyBJacobs Other endpoints, allow this pattern. 

https://github.com/WordPress/wordpress-develop/blob/9cc20468ef4803510f24caee93082514224f01f2/src/wp-includes/rest-api/endpoints/class-wp-rest-terms-controller.php#L170

https://github.com/WordPress/wordpress-develop/blob/9cc20468ef4803510f24caee93082514224f01f2/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L501-L505

Why make different in this endpoint?

Reply:
Returning boolean true is fine, but if we return a boolean false then the REST API just returns a generic auth permission error. Returning a `WP_Error` instance let's us give a more clear error message to users and a specific error code.

I think ideally we should make sure endpoints are returning proper error objects when possible.

Reply:
Updated. Can you review again @TimothyBJacobs ?

Reply:
Resolved in 1a7f3d7bea8166bd82d0a72551dcf89cf8fa9b05 👍